### PR TITLE
[Community] Add './projects' folder and an example

### PR DESCRIPTION
**Pull request by fangyixiao18**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Refer to https://github.com/open-mmlab/mmocr/pull/1524

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: projects/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Welcome to Projects of MMSelfSup
+
+In this folder, we welcome all contribution of self-supervised learning algorithms from community.
+
+Here, the requirements are not that strict in core package. Thus, it is able to implement algorithms efficiently and contribute to MMSelfSup. We appreciate all contributors to make MMSelfSup greater.
```
Comment (YuanLiuuuuuu) [inline projects/README.md]:
```suggestion
Here, these requirements, e.g. code standards, are not that strict as in core package. Thus, developers from the community can implement their algorithms much more easily and efficiently in MMSelfSup. We appreciate all contributions from community to make MMSelfSup greater.
```

File: projects/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+For more details, we provide some documentation listed below:
```
Comment (YuanLiuuuuuu) [inline projects/README.md]:
```suggestion
We give some instructions to implement your algorithms under this folder:
```

File: projects/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+- [Example Project README](./example_project/README.md)
+
+  The explanation and milestons for example projects.
```
Comment (YuanLiuuuuuu) [inline projects/README.md]: This line is confused. Try to make it more clear.

Comment (fangyixiao18) [reply projects/README.md]: I re-organized this part

**Review by tonysy — APPROVED** [review]
LGTM

**Review by YuanLiuuuuuu — APPROVED** [review]

**Pull request merged**